spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/BlasWrapper.scala

```diff
@@ -69,10 +84,82 @@ private[bigdl] class BlasWrapper(val module: AbstractModule[Activity, Activity,
     }).toArray
   }
 
+  /**
+   * Blas layers may not have good performance with mkldnn backend,
+   * so we can use java multi thread to run blas layers and thread number is
+   * related with batch size and core number.
```

Comment:
=> we can leverage multi-threading to resolve bottleneck brought by one model only for dnn,  the parallelism is determined by both bath size and core number, with restrictions that both input and output format must be batched